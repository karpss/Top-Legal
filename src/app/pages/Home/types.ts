import { FetchCharactersList } from "../../services/charactersService/__generated__/FetchCharactersList";

export interface IHomeState {
  charactersList: FetchCharactersList["characters"];
}
