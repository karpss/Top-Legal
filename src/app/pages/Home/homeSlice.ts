import { createSlice } from "@reduxjs/toolkit";
import { IHomeState } from "./types";

const initialState: IHomeState = {
  charactersList: null,
};

const HomeSlice = createSlice({
  name: "home",
  initialState,
  reducers: {
    setCharactersList(state, action) {
      const ref = state;
      ref.charactersList = action.payload;
    },
  },
});

export const { setCharactersList } = HomeSlice.actions;
export default HomeSlice.reducer;
