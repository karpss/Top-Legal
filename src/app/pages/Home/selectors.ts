import { createSelector } from "reselect";
import { IRootState } from "../../types";

const selectHome = (state: IRootState) => state.home;

const makeSelectCharactersList = createSelector(
  selectHome,
  (home) => home.charactersList,
);

export default makeSelectCharactersList;
