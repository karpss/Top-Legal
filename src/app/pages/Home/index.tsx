import React, { useCallback, useEffect } from "react";
import { Dispatch } from "redux";
import charactersService from "../../services/charactersService";
import { useAppDispatch } from "../../hooks";
import { FetchCharactersList } from "../../services/charactersService/__generated__/FetchCharactersList";
import { setCharactersList } from "./homeSlice";
import CardDetails from "../../components";

const actionDispatch = (dispatch: Dispatch) => ({
  setCharactersList: (page: FetchCharactersList["characters"]) =>
    dispatch(setCharactersList(page)),
});

function Home() {
  // eslint-disable-next-line @typescript-eslint/no-shadow
  const { setCharactersList } = actionDispatch(useAppDispatch());

  const fetchCharacters = useCallback(async () => {
    const charactersList = await charactersService
      .fetchCharactersList(0)
      .catch((err) => {
        return err;
      });
    if (charactersList) {
      setCharactersList(charactersList);
    }
  }, [setCharactersList]);

  useEffect(() => {
    fetchCharacters();
  }, [fetchCharacters]);

  return (
    <div>
      <CardDetails />
    </div>
  );
}

export default Home;
