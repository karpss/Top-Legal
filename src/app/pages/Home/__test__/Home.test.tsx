import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "../../../store";
import Home from "../index";

describe("Home Component", () => {
  it("should render home component", () => {
    const { container } = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
