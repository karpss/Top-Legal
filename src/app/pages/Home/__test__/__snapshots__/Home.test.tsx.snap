// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component should render home component 1`] = `
<div>
  <div>
    <div>
      <div
        class="MuiBox-root css-8atqhb"
      >
        <span
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate css-eglki6-MuiLinearProgress-root"
          role="progressbar"
        >
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate css-5ir5xx-MuiLinearProgress-bar1"
          />
          <span
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar2Indeterminate css-1r8wrcl-MuiLinearProgress-bar2"
          />
        </span>
      </div>
    </div>
  </div>
</div>
`;
